include/wlr/types/wlr_xdg_shell.h

```diff
@@ -121,10 +125,11 @@ struct wlr_xdg_toplevel {
 	struct wlr_xdg_surface *parent;
 	struct wl_listener parent_unmap;
 
-	struct wlr_xdg_toplevel_state client_pending;
-	struct wlr_xdg_toplevel_state server_pending;
-	struct wlr_xdg_toplevel_state last_acked;
-	struct wlr_xdg_toplevel_state current;
+	struct wlr_xdg_toplevel_state current, pending;
+	struct wlr_xdg_toplevel_state scheduled;
+
+	// Intended to be checked on map to handle events before it
```

Comment:
This comment could be a bit more clear that the compositor author using wlroots is the one intended to check this, as opposed to the check being done internally to wlroots.

Reply:
Added a hopefully more clear comment.